inject/src/main/java/io/micronaut/context/converters/StringToCharArray.java

```diff
@@ -0,0 +1,16 @@
+package io.micronaut.context.converters;
+
+import io.micronaut.core.convert.ConversionContext;
+import io.micronaut.core.convert.TypeConverter;
+
+import javax.inject.Singleton;
+import java.util.Optional;
+
+@Singleton // #2614
```

Comment:
I don't believe this is working as intended because beans cannot be defined in the `inject` module since that needs to be compiled first. Also see @jameskleeh comment